Importing archive cause SQL error and in failure

When trying to reimport an archive on an existing fresh installation, it fails even if the archive is the right version.

**Seen behavior**
In error logs, the following error is displayed : 
`Traceback (most recent call last):
  File "C:\Users\jb\Libreosteo\libreosteoweb\api\views.py", line 775, in post
  File "C:\Users\jb\Libreosteo\venv_64\lib\site-packages\django\core\management\__init__.py", line 148, in call_command
  File "C:\Users\jb\Libreosteo\venv_64\lib\site-packages\django\core\management\base.py", line 364, in execute
  File "C:\Users\jb\Libreosteo\venv_64\lib\site-packages\django\core\management\commands\flush.py", line 65, in handle
django.core.management.base.CommandError: Database C:\Program Files (x86)\Libreosteo\db.sqlite3 couldn't be flushed. Possible reasons:
  * The database isn't running or isn't configured correctly.
  * At least one of the expected database tables doesn't exist.
  * The SQL was invalid.
Hint: Look at the output of 'django-admin sqlflush'. That's the SQL this command wasn't able to run.

2022-01-13 21:54:19,602 [ERROR] libreosteoweb.api.views: Import failed
Traceback (most recent call last):
  File "C:\Users\jb\Libreosteo\venv_64\lib\site-packages\django\db\backends\utils.py", line 82, in _execute
  File "C:\Users\jb\Libreosteo\venv_64\lib\site-packages\django\db\backends\sqlite3\base.py", line 381, in execute
sqlite3.IntegrityError: FOREIGN KEY constraint failed`

**Analysis**

When the flush is performed the following sequence is executed, but dependings of the execution the order is not always the same : 
`BEGIN;
DELETE FROM "libreosteoweb_officeevent";
DELETE FROM "libreosteoweb_fileimport";
DELETE FROM "zipcode_lookup_zipcodemapping";
DELETE FROM "django_admin_log";
DELETE FROM "libreosteoweb_document";
DELETE FROM "auth_user";
DELETE FROM "auth_user_user_permissions";
DELETE FROM "django_content_type";
DELETE FROM "libreosteoweb_patientdocument";
DELETE FROM "libreosteoweb_invoice";
DELETE FROM "auth_permission";
DELETE FROM "libreosteoweb_examinationcomment";
DELETE FROM "libreosteoweb_patient";
DELETE FROM "libreosteoweb_therapeutsettings";
DELETE FROM "libreosteoweb_officesettings";
DELETE FROM "libreosteoweb_regulardoctor";
DELETE FROM "auth_group";
DELETE FROM "django_session";
DELETE FROM "libreosteoweb_examination";
DELETE FROM "auth_user_groups";
DELETE FROM "libreosteoweb_paiment_invoice";
DELETE FROM "libreosteoweb_examination_invoices";
DELETE FROM "libreosteoweb_paimentmean";
DELETE FROM "auth_group_permissions";
DELETE FROM "libreosteoweb_paiment";
DELETE FROM "libreosteoweb_loggedinuser";
DELETE FROM "libreosteoweb_children";
COMMIT;`

But there is a foreign key on django_content_types to django_admin_log and auth_permission.
The delete does not work, flush in failure

--- a/libreosteoweb/api/views.py
+++ b/libreosteoweb/api/views.py
@@ -48,6 +48,7 @@ from django.views import View
 from django.views.decorators.cache import never_cache
 from django.views.generic.base import TemplateView
 from django.db.models import Max
+from django.db import connection
 from libreosteoweb.api import serializers as apiserializers
 from libreosteoweb import models
 from django.contrib.auth.mixins import PermissionRequiredMixin
@@ -68,6 +69,7 @@ from libreosteoweb.api.events.settings import settings_event_tracer, full_db_dow
 from django.core.files.storage import default_storage
 from libreosteoweb.api.signals import post_reload_db
 import uuid
+from io import StringIO
 
 # Get an instance of a logger
 logger = logging.getLogger(__name__)
@@ -333,7 +335,7 @@ class ExaminationViewSet(viewsets.ModelViewSet):
             return
 
         if serializer.validated_data['date'] and serializer.instance and serializer.instance.last_invoice \
-                                            and serializer.validated_data['date'] < serializer.instance.last_invoice.date :
+                and serializer.validated_data['date'] < serializer.instance.last_invoice.date:
             return
         raise SuspiciousOperation(
             "Invalid request : examination date is not allowed")
@@ -445,7 +447,8 @@ class InvoiceViewSet(viewsets.ReadOnlyModelViewSet):
 
                     result = invoicing_generator.ExaminationInvoiceHelper(officesettings, therapeutsettings, request.user)\
                         .invoice_examination(invoicing_serializer,
-                                             models.Examination.objects.get(id=request.data['examination']['id']),
+                                             models.Examination.objects.get(
+                                                 id=request.data['examination']['id']),
                                              self.get_object())
                     if 'errors' in result:
                         return Response(result['errors'],
@@ -740,7 +743,7 @@ class LoadDump(View):
                                 'This file is an archive of the version {otherversion}, the current version is {currentversion}. Install the version {otherversion} and load it.',
                                 otherversion=v,
                                 currentversion=libreosteoweb.__version__),
-                                                status=412)
+                                status=412)
                     # uncompress the dump file
                     if filename in zf.namelist():
                         zf.extract(filename, tmpdir)
@@ -772,9 +775,19 @@ class LoadDump(View):
                     logger.info(
                         "Signals were disactivated, perform clearing of the database"
                     )
-                    call_command('flush',
-                                 interactive=False,
-                                 stdout=LoggerWriter(logger.info))
+                    buf = StringIO()
+                    call_command('sqlflush', no_color=True, stdout=buf)
+                    with connection.cursor() as cursor:
+                        deferred_delete = ''
+                        for s in buf.getvalue().split('\n'):
+                            if "django_content_type" not in s and "COMMIT" not in s:
+                                logger.info("Execute query : %s" % s)
+                                cursor.execute(s)
+                            else:
+                                deferred_delete = deferred_delete + '\n' + s
+                        for s in deferred_delete.split('\n'):
+                            logger.info(s)
+                            cursor.execute(s)
                     # It means that the settings.FIXTURE_DIRS should be set in settings
                     previous = settings.FIXTURE_DIRS
                     settings.FIXTURE_DIRS = [tempfile.gettempdir()]
@@ -799,4 +812,4 @@ class LoadDump(View):
             return HttpResponse(content=_(
                 u'This archive file seems to be incorrect. Impossible to load it.'
             ),
-                                status=412)
+                status=412)

